main.tf

```diff
@@ -4,15 +4,30 @@ locals {
   s3_arn_prefix = "arn:${one(data.aws_partition.default[*].partition)}:s3:::"
 
   is_vpc = var.vpc_id != null
-
-  user_names = keys(var.sftp_users)
-
+  # create map that include the data we need for creating sftp user, adding the s3 bucket to the sftp_users variable
   user_names_map = {
-    for user, val in var.sftp_users :
-    user => merge(val, {
+    for val in var.sftp_users :
+    val.user_name => merge(val, {
       s3_bucket_arn = lookup(val, "s3_bucket_name", null) != null ? "${local.s3_arn_prefix}${lookup(val, "s3_bucket_name")}" : one(data.aws_s3_bucket.landing[*].arn)
     })
   }
+  # create list of maps that holds the public keys of each user, in that way we can have more than one public key to user
+  ssh_keys = flatten([
+    for val in var.sftp_users : [
+      for key in val["public_keys"] : {
+        user_name  = val["user_name"]
+        public_key = key,
+        token      = "${val["user_name"]}#${key}"
```

Comment:
These tokens could end up being really long. I'm unsure how Terraform / Tofu handles extremely long resource location keys. I'm of the mind that we hash this value using [`md5`](https://developer.hashicorp.com/terraform/language/functions/md5) to ensure that these values do not cause any issues with length requirements. Can you update to that?